### PR TITLE
Archetypes for other types of plugins

### PR DESCRIPTION
**Pull request by mtrojanowski**
Archetypes for all the other types of plug-ins.

### CODE REVIEW

File: claims-provider-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__ClaimsProvider.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+        // See examples of Claims Providers on GitHub: https://github.com/search?q=topic%3Aclaims-provider+org%3Acurityio
+    }
+
+    @Override
+    public Attributes getClaimValues(Set<RequestedClaimAttribute> requestedClaimAttributes,
+        Set<ScopeValue> scopeValues,
+        AuthenticationAttributes userAuthenticationAttributes,
+        OAuthClient client)
+    {
```
Comment (travisspencer) [inline claims-provider-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__ClaimsProvider.java:28]: we should be consistent in formatting; we don't have to use our style necessarily, but we shouldn't mix styles within the same file

File: consentor-archetype/src/test/resources/projets/basic/archetype.properties [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package=com.example.curity.extensions
+version=1.0.0-SNAPSHOT
+groupId=com.example.curity
+artifactId=example-curity-hello-world-consentor
+pluginName=HelloWorldConsentor
+curitySdkVersion=5.0.0
```
Comment (travisspencer) [inline consentor-archetype/src/test/resources/projets/basic/archetype.properties]: new lines should be added. Without, things like `cat`, produce harder-to-use or wrong results

File: data-access-provider-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__DeviceDataAccessProvider.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import se.curity.identityserver.sdk.attribute.scim.v2.ResourceAttributes;
+import se.curity.identityserver.sdk.attribute.scim.v2.extensions.DeviceAttributes;
+import se.curity.identityserver.sdk.data.query.ResourceQuery;
+import se.curity.identityserver.sdk.data.query.ResourceQueryResult;
+import se.curity.identityserver.sdk.datasource.DeviceDataAccessProvider;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+import java.util.List;
+
+public class ${pluginName}DeviceDataAccessProvider implements DeviceDataAccessProvider
```
Comment (travisspencer) [inline data-access-provider-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__DeviceDataAccessProvider.java]: final

File: data-access-provider-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__DynamicallyRegisteredClientDataAccessProvider.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
+package ${package}.${pluginName};
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import se.curity.identityserver.sdk.attribute.scim.v2.extensions.DynamicallyRegisteredClientAttributes;
+import se.curity.identityserver.sdk.datasource.DynamicallyRegisteredClientDataAccessProvider;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+public class ${pluginName}DynamicallyRegisteredClientDataAccessProvider implements DynamicallyRegisteredClientDataAccessProvider
```
Comment (travisspencer) [inline data-access-provider-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__DynamicallyRegisteredClientDataAccessProvider.java]: etc.

File: data-access-provider-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__UserAccountDataAccessProvider.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import se.curity.identityserver.sdk.attribute.AccountAttributes;
+import se.curity.identityserver.sdk.attribute.scim.v2.ResourceAttributes;
+import se.curity.identityserver.sdk.attribute.scim.v2.extensions.LinkedAccount;
+import se.curity.identityserver.sdk.data.query.ResourceQuery;
+import se.curity.identityserver.sdk.data.query.ResourceQueryResult;
+import se.curity.identityserver.sdk.data.update.AttributeUpdate;
+import se.curity.identityserver.sdk.datasource.UserAccountDataAccessProvider;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
```
Comment (travisspencer) [inline data-access-provider-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__UserAccountDataAccessProvider.java]: you shouldn't use this. Use java.lang.UnsupportedOperationException

**Review by travisspencer — CHANGES_REQUESTED** [review]

File: claims-provider-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__ClaimsProvider.java (hunk 1 above)
Comment (mtrojanowski) [reply claims-provider-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__ClaimsProvider.java:28]: It's hard to style automatically those files as IntelliJ doesn't recognise properly the template variables used in method or class names. I'll try to go once again through the files and change formatting.

Comment (mtrojanowski) [reply claims-provider-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__ClaimsProvider.java:28]: The formatting should now be consistent.

**Review by travisspencer — APPROVED** [review]

File: claims-provider-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__ClaimsProvider.java (hunk 1 above)
Comment (travisspencer) [reply claims-provider-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__ClaimsProvider.java:28]: Understandable. It'll be easy for the author to reformat as well, so let's not loose sleep over this.

**Pull request merged**